salt/states/archive.py

```diff
@@ -823,17 +834,20 @@ def extracted(name,
         )
         return ret
 
-    if trim_output and not isinstance(trim_output, (bool, six.integer_types)):
-        try:
-            # Try to handle cases where trim_output was passed as a
-            # string-ified integer.
-            trim_output = int(trim_output)
-        except TypeError:
-            ret['comment'] = (
-                'Invalid value for trim_output, must be True/False or an '
-                'integer'
-            )
-            return ret
+    if trim_output:
+        if trim_output is True:
+            trim_output = 100
+        elif not isinstance(trim_output, (bool, six.integer_types)):
+            try:
+                # Try to handle cases where trim_output was passed as a
+                # string-ified integer.
+                trim_output = int(trim_output)
+            except TypeError:
+                ret['comment'] = (
+                    'Invalid value for trim_output, must be True/False or an '
+                    'integer'
+                )
+                return ret
```

Comment:
@terminalmage I'm just thinking maybe it is easier to comprehend by first discarding `bool`?
```python
trim_output = 100 if type(trim_output) == bool and trim_output else False
if trim_output and type(trim_output) != bool:
    try:
        trim_output = int(trim_output)
    except TypeError:
        ...
```